pkg/util/filewatcher/filewatcher.go

```diff
@@ -0,0 +1,65 @@
+package filewatcher
+
+import (
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+	"k8s.io/klog/v2"
+)
+
+var watchCertificateFileOnce sync.Once
+
+// WatchFileForChanges watches the file, fileToWatch, for changes. If the file contents have changed, the pod this
+// function is running on will be restarted.
+func WatchFileForChanges(fileToWatch string) error {
```

Comment:
I had. I was actually planning to put in a follow up PR in librarygo with this function and take care of it there. After that PR, I was planning to update ingress, cncc, and image registry operators with that function so that the exact same function was being utilized by all 3. 

There wasn't enough time to do that initially. We need to get this into 4.18 and backported to 4.17.z before the ARO HCP milestone P2 which is Dec 13th.